pkg/openstack/loadbalancer.go

```diff
@@ -1204,19 +1204,22 @@ func (lbaas *LbaasV2) ensureOctaviaHealthMonitor(lbID string, name string, pool
 
 // buildMonitorCreateOpts returns a v2monitors.CreateOpts without PoolID for consumption of both, fully popuplated Loadbalancers and Monitors.
 func (lbaas *LbaasV2) buildMonitorCreateOpts(svcConf *serviceConfig, port corev1.ServicePort) v2monitors.CreateOpts {
-	monitorProtocol := string(port.Protocol)
-	if port.Protocol == corev1.ProtocolUDP {
-		monitorProtocol = "UDP-CONNECT"
-	}
-	if svcConf.healthCheckNodePort > 0 {
-		monitorProtocol = "HTTP"
-	}
-	return v2monitors.CreateOpts{
-		Type:       monitorProtocol,
+	opts := v2monitors.CreateOpts{
+		Type:       string(port.Protocol),
 		Delay:      svcConf.healthMonitorDelay,
 		Timeout:    svcConf.healthMonitorTimeout,
 		MaxRetries: svcConf.healthMonitorMaxRetries,
 	}
+	if port.Protocol == corev1.ProtocolUDP {
+		opts.Type = "UDP-CONNECT"
+	}
+	if svcConf.healthCheckNodePort > 0 {
+		opts.Type = "HTTP"
+		opts.URLPath = "/healthz"
```

Comment:
This /healthz is used for pod healthy status check
but you are proposing to do this for openstack? Not against it but want to make sure I understand the reason
and how its used..

Reply:
This is the health check on the nodes. So, if a load balancer service in Kubernetes has `healthCheckNodePort` set, this port can be used to check if the service on the node port is healthy or not. When creating a health monitor in Openstack, the load balancer uses this port to check, if the service is healthy on the node.

This seems to have worked fine with a request to `/`, but as you can see in the linked Kubernetes code, `/healthz` is the preferred endpoint here.

Reply:
got it ,thanks for the detailed info~